src/web/app/components/student-list/student-list.component.ts

```diff
@@ -11,16 +15,11 @@ import { JoinState, MessageOutput } from '../../../types/api-output';
 import { SortBy, SortOrder } from '../../../types/sort-properties';
 import { ErrorMessageOutput } from '../../error-message-output';
 import { JoinStatePipe } from './join-state.pipe';
-import {
-  StudentListSectionData,
-  StudentListStudentData,
-} from './student-list-section-data';
 
 /**
- * Flattened data which contains details about a student and their section.
- * The data is flattened to allow sorting of the table.
+ * Model of row of student data containing details about a student and their section.
  */
-interface FlatStudentListData {
+export interface StudentListRowModel {
```

Comment:
For your question:

I think you can actually define a new field: `student: Student` here (and delete other student related field). Store the model in the corresponding page and when stats is calculated. Use the `student` field to construct a list of student.